packages/components/organisms/f-checkout/src/store/checkout.module.js

```diff
@@ -22,6 +22,26 @@ import {
 } from './mutation-types';
 import checkoutIssues from '../checkout-issues';
 
+/**
+* @function resolveCustomerDetails
+* If certain customer details are missing from api `data.customer` object then fallback
+* to the decoded `state.AuthToken` details and re-assign back to `data.customer`.
+* @param  {object} data  - Api response object.
+* @param  {object} state - The current `checkout` state.
+*/
+function resolveCustomerDetails (data, state) {
+    const tokenData = jwtDecode(state.authToken);
```

Comment:
It might be premature optimization, but in the replaced code, the token was only decoded if the necessary keys (previously only `phoneNumber`) weren't set in the given data. Is it worth doing the same here?